### PR TITLE
BS-14942 store client-bdm.zip in the TenantResourcesService

### PR DESCRIPTION
**Pull request by baptistemesta**
depends on https://github.com/bonitasoft/bonita-engine/pull/259

Store it in database using the TenantResourcesService instead of having
it in the FS in the bonita home

### CODE REVIEW

File: bpm/bonita-core/bonita-home-server/src/main/java/org/bonitasoft/engine/home/TenantStorage.java [hunk 1]
```diff
@@ -36,32 +35,6 @@ public TenantStorage(BonitaHomeServer bonitaHomeServer) {
         this.bonitaHomeServer = bonitaHomeServer;
     }
 
-    private File getBDMFile(final long tenantId) throws BonitaHomeNotSetException, IOException {
-        final Folder bdmFolder = FolderMgr.getTenantWorkBDMFolder(getBonitaHomeFolder(), tenantId);
```
Comment (educhastenier) [inline bpm/bonita-core/bonita-home-server/src/main/java/org/bonitasoft/engine/home/TenantStorage.java]: You could also remove unused method FolderMgr.getTenantWorkBDMFolder()

File: services/bonita-business-data/bonita-business-data-impl/src/main/java/org/bonitasoft/engine/business/data/impl/BusinessDataModelRepositoryImpl.java [hunk 2]
```diff
@@ -72,22 +74,27 @@
...
+        STenantResource sTenantResource;
         try {
-            return BonitaHomeServer.getInstance().getTenantStorage().getClientBDMZip(tenantId);
-        } catch (final Exception e) {
+            sTenantResource = tenantResourcesService.get(TenantResourceType.BDM, CLIENT_BDM_ZIP);
+        } catch (SBonitaReadException e) {
             throw new SBusinessDataRepositoryException(e);
         }
+        if (sTenantResource == null) {
+            throw new SBusinessDataRepositoryException("no client-bdnm.zip found in tenant resources");
```
Comment (educhastenier) [inline services/bonita-business-data/bonita-business-data-impl/src/main/java/org/bonitasoft/engine/business/data/impl/BusinessDataModelRepositoryImpl.java]: typo

File: services/bonita-business-data/bonita-business-data-impl/src/test/java/org/bonitasoft/engine/business/data/impl/BusinessDataModelRepositoryImplTest.java [hunk 3]
```diff
@@ -81,6 +69,16 @@ public void deployABOMShouldCreatetheBOMJARAndAddANewTenantDependency() throws E
     }
 
     @Test
+    public void deployABOMShouldCreatetheShouldDeployTenantDepenencyWithClientZip() throws Exception {
```
Comment (educhastenier) [inline services/bonita-business-data/bonita-business-data-impl/src/test/java/org/bonitasoft/engine/business/data/impl/BusinessDataModelRepositoryImplTest.java]: review name

**Comment by educhastenier** [comment]
Looks good to me

File: bpm/bonita-core/bonita-home-server/src/main/java/org/bonitasoft/engine/home/TenantStorage.java (hunk 1 above)
Comment (baptistemesta) [inline bpm/bonita-core/bonita-home-server/src/main/java/org/bonitasoft/engine/home/TenantStorage.java]: done

**Comment by baptistemesta** [comment]
Merge after fix of review

**Pull request merged**